client/consul_template.go

```diff
@@ -384,15 +385,13 @@ func parseTemplateConfigs(tmpls []*structs.Template, taskDir string,
 			dest = filepath.Join(taskDir, taskEnv.ReplaceEnv(tmpl.DestPath))
 		}
 
-		ct := ctconf.ConfigTemplate{
-			Source:           src,
-			Destination:      dest,
-			EmbeddedTemplate: tmpl.EmbeddedTmpl,
-			Perms:            ctconf.DefaultFilePerms,
-			Wait:             &watch.Wait{},
-		}
+		ct := ctconf.DefaultTemplateConfig()
+		ct.Source = &src
```

Comment:
There are `config` helpers that assist in taking the address of things:

```go
config.String("foo")
```

but this works too